api/controllers/v0.2/createPatron.js

```diff
@@ -55,74 +55,155 @@ function renderResponse(req, res, status, message) {
 }
 
 /**
- * createPatron(req, res)
- * The callback for the route "/patrons".
- * It will fire a POST request to the ILS API for creating a new patron.
+ * streamPatron(req, res, streamPatron, modeledResponse)
+ * Stream the new patron data to Kinesis
  *
  * @param {HTTP request} req
  * @param {HTTP response} res
+ * @param {object} streamPatron
+ * @param {object} modeledResponse
  */
-function createPatron(req, res) {
-  if (!process.env.ILS_CREATE_TOKEN_URL || isEmpty(process.env.ILS_CREATE_TOKEN_URL)) {
-    res
-      .status(500)
-      .header('Content-Type', 'application/json')
-      .json(modelResponse.errorResponseData(
-        collectErrorResponseData(
-          null,
-          'invalid-request',
-          'The create-patron service is missing the ILS_CREATE_TOKEN_URL.',
-          null,
-          { form: ['The create-patron service is misconfigured.'] } // eslint-disable-line comma-dangle
-        ) // eslint-disable-line comma-dangle
-      ));
+function streamPatron(req, res, streamPatronData, modeledResponse) {
+  streamPublish.streamPublish(
+    process.env.PATRON_SCHEMA_NAME_V02,
+    process.env.PATRON_STREAM_NAME_V02,
+    streamPatronData // eslint-disable-line comma-dangle
+  ).then((streamResponse) => { // eslint-disable-line no-unused-vars
+    renderResponse(req, res, 201, modeledResponse);
+    console.log('Published to stream successfully!'); // eslint-disable-line no-console
+  }).catch((streamError) => {
+    renderResponse(req, res, 201, streamError);
+    console.error(`Error publishing to stream: ${streamError}`); // eslint-disable-line no-console
+  });
+}
 
-    return;
-  }
+/**
+ * callAxiosToCreatePatron(req, res, tokenResponse)
+ * Stream the new patron data to Kinesis
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ * @param {object} tokenResponse
+ */
+function callAxiosToCreatePatron(req, res, tokenResponse) {
+  axios.post(process.env.ILS_CREATE_PATRON_URL, req.body, {
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${tokenResponse.data.access_token}`,
+    },
+  })
+    .then((response) => {
+      const modeledResponse =
+        modelResponse.patronCreator(response.data, response.status, req.body);
+      modelStreamPatron.transformPatronRequest(
+        req.body, modeledResponse // eslint-disable-line comma-dangle
+      )
+        .then((streamPatronData) => {
+          streamPatron(req, res, streamPatronData, modeledResponse);
+        })
+        .catch((error) => {
+          renderResponse(req, res, 201, modeledResponse);
```

Comment:
What kind of error would be caught here? Seems like an error writing to the stream would be caught inside `streamPatron`, so this is only catching errors thrown by `modelStreamPatron.transformPatronRequest`, which should resolve almost unconditionally. If there are known types of errors caught here, is there a reason you'd want to render `modeledResponse` with http status 201 in those cases? In other words, we can only reach this `catch` block if an error was thrown in `modelStreamPatron.transformPatronRequest`, so it's odd to communicate that the patron record was created.

Reply:
Righto!